src/AppHost.ts

```diff
@@ -1,73 +1,69 @@
 export interface IAppHost {
     /**
-     * Generates a profile of media the app can stream
+     * Queries the host application for the name of the app.
      *
-     * @param profileBuilder A helper that makes the profile creation easier
+     * @returns The name of the client.
      */
-    getDeviceProfile(profileBuilder: unknown): Promise<unknown>;
+    appName(): string;
 
     /**
-     * Generates a profile of media the app can sync
+     * Queries the host application for a the version of the app.
      *
-     * @param profileBuilder A helper that makes the profile creation easier
-     * @param appSettings TODO: Add description
-     */
-    getSyncProfile(
-        profileBuilder: unknown,
-        appSettings: unknown
-    ): Promise<unknown>;
-
-    /**
-     * Called when the app exits
+     * @returns The version of the app.
      */
-    exit(): void;
+    appVersion(): string;
 
     /**
-     * Return a boolean value indicating if a command is supported in the app.
+     * Queries the host application for a unique Device ID.
      *
-     * _For supported commands check: {@link Command}_
-     *
-     * @param command command that should be checked
+     * @returns The unique ID of the device.
      */
-    supports(command: Command): boolean;
+    deviceID(): string;
 
     /**
-     * Return the default UI layout
+     * Queries the host application for a display name from the device.
+     *
+     * @returns The display name of the device.
      */
-    getDefaultLayout(): Layout;
+    deviceName(): string;
 
     /**
-     * Returns basic info about the app
+     * Dispatches exit procedures to the host application.
      *
-     * _Called once startup_
+     * _Currently only supported on Tizen._
      */
-    init(): IAppInfo;
+    exit(): void;
 
     /**
-     * A unique and persistent device identifier
+     * Queries the host application for the default layout to use in the client.
+     *
+     * @returns A string containing the identifier of the layout to use.
      */
-    deviceID(): string;
+    getDefaultLayout(): Layout;
 
     /**
-     * A display name from the device
+     *  Queries the host application for a profile of media that are natively supported by the app.
      *
-     * _Example: My Phone_
+     * @param profile A Profile object generated by the web client and describing the media capabilities of the client.
+     * @returns A Profile object describing the full capabilities of the client.
      */
-    deviceName(): string;
+    getDeviceProfile(profile: IProfile): IProfile;
```

Comment:
Why is the profile builder gone here? I've seen it used in the wild

Reply:
It's not a profile builder, but the return of that profile builder, which is a profile object (That is documented further down in the file).

The module that is passed to it in the web client is https://github.com/jellyfin/jellyfin-web/blob/master/src/scripts/browserdeviceprofile.js

If you notice, the `return` of that module (which is what is passed to getDeviceProfile) generated a profile.

Reply:
https://github.com/jellyfin/jellyfin-android/blob/14f928b3396bc60c138dbed108965020924f52c2/src/NativeShell/www/apphost.js#L22

Reply:
Seems like Android is different than what was intended to be documented by this: https://github.com/jellyfin/jellyfin-web/blob/master/src/components/apphost.js#L47

Reply:
Well that makes the type still wrong